docs/timezone.md

```diff
@@ -133,12 +135,67 @@ tz2 = Temporal.TimeZone.from(tz);
 ### timeZone.**id** : string
 
 The `id` property gives an unambiguous identifier for the time zone.
-Effectively, this is the canonicalized version of whatever `timeZoneIdentifier` was passed as a parameter to the constructor.
+This is the normalized version of whatever `timeZoneIdentifier` was passed as a parameter to the constructor.
 
 When subclassing `Temporal.TimeZone`, this property must be overridden to provide an identifier for the custom time zone.
 
 ## Methods
 
+### timeZone.**equals**(_other_: Temporal.TimeZone | object | string) : boolean
+
+**Parameters:**
+
+- `other` (`Temporal.TimeZone` object, object implementing the `Temporal.TimeZone` protocol, or a string time zone identifier): Another time zone to compare.
+
+**Returns:** `true` if `timeZone` and `other` are equal, or `false` if not.
+
+Compares two time zones for equality.
+Equality is determined by the following algorithm:
+
+- If `timeZone === other`, then the time zones are equal.
+- Otherwise, `timeZone.id` is compared to `other` (or `other.id` if `other` is an object).
+  If both string identifiers are equal, then the time zones are equal.
+  This is a case-sensitive comparison; custom time zone identifiers that vary in case will be considered unequal.
+- The time zones are also equal if the identifiers resolve to the same Zone name in the [IANA Time Zone Database](https://www.iana.org/time-zones).
+  This resolution is case-insensitive.
+- If both identifiers are numeric offset time zone identifiers like `+05:30`, then they are equal if they represent the same offset.
+- Otherwise, the time zones are not equal.
+
+Time zones that resolve to different Zones in the IANA Time Zone Database are not equal, even if those Zones use the same offsets.
+Similarly, a numeric-offset identifier is never equal to a named time zone in the IANA Time Zone Database, even if they represent the same offsets.
+
+Although there may be slight variation between implementations, ECMAScript implementations generally build the IANA Time Zone Database using build options that guarantee at least one Zone for every <a href="https://www.iso.org/glossary-for-iso-3166.html">ISO 3166-1 Alpha-2</a> country code.
+This behavior differs from the default build options of the IANA Time Zone Database where a Zone may span multiple countries that have shared the same offsets since 1970, for example Europe/Oslo, Europe/Stockholm, Europe/Copenhagen, and Europe/Berlin.
+To avoid conflating different countries' time zones which may vary in the future, ECMAScript does not use these default build options.
```

Comment:
This seems too strong, I don't think we established it as a normative requirement.

Reply:
Yeah, I wasn't sure how to talk about the case where there's consistent behavior among implementations that nonetheless isn't in the spec. And especially because this implementation-defined behavior is very unlikely to change. Got a suggestion for how to thread the needle?

The main thing I wanted to avoid was users going to the [wikipedia page](https://en.wikipedia.org/wiki/List_of_tz_database_time_zones) (which uses the default TZDB build options) and being confused that IDs shown as Links there aren't considered non-primary in ECMAScript. No implementation treats Europe/Oslo or Atlantic/Reykjavik as non-primary, but Wikipedia does.

Although there's well-known variation between engines about which ID is canonical among a group of IDs (e.g. Asia/Kolkata in FF vs. Asia/Calcutta in Chrome), the groups of IDs themselves are very consistent across implementations. The most populous zone where they disagree is Slovakia vs. Czechia whose IDs are considered equal in FF but different in Chrome. The other differences are really small countries like Europe/Vatican where FF considers a Link to Europe/Rome but (because it's a different country code) CLDR and hence Chrome/Safari do not.

I'm trying to fix this variation in https://github.com/tc39/ecma402/issues/806, and if we can successfully get FF to change to match the same grouping as Chrome/Safari, then I think we can indeed make the text above normative.

So the main question to me is what to say in the meantime.

Reply:
This was resolved via a suggestion on another thread after a rebase crossed GitHub's wires.